pkg/base1/test-timeformat.ts

```diff
@@ -0,0 +1,150 @@
+    assert.equal(timeformat.date(d1), "2 de janeiro de 2024");
+
+    cockpit.language = "pt_BR";
+    assert.equal(timeformat.date(d1), "2 de janeiro de 2024");
+
+    cockpit.language = "zh_CN";
+    assert.equal(timeformat.weekdayDate(d1), "2024年1月2日星期二");
+
+    cockpit.language = "zh_TW";
+    assert.equal(timeformat.weekdayDate(d1), "2024年1月2日 星期二");
+});
+
+const SEC = 1000;
+const MIN = 60 * SEC;
+const HOUR = 60 * MIN;
+const DAY = 24 * HOUR;
+
+QUnit.test("relative formatter, English", assert => {
+    const now = Date.now();
+    const seconds = now - 4.5 * SEC;
+    const minutes = now - 4 * MIN - 5 * SEC;
+    const halfhour = now - 32 * MIN;
+    const hours = now - 3 * HOUR - 57 * MIN;
+    const days4short = now - 4 * DAY + 2 * HOUR;
+    const days4long = now - 4 * DAY - 2 * HOUR;
+    const week3short = now - 20 * DAY + 2 * HOUR;
+    const week3long = now - 21 * DAY - 2 * HOUR;
+    const month2short = now - 2 * 30 * DAY + 5 * DAY;
+    const month2long = now - 2 * 30 * DAY - 5 * DAY;
+
+    cockpit.language = "en";
+    assert.equal(timeformat.distanceToNow(seconds), "less than a minute");
+    assert.equal(timeformat.distanceToNow(seconds, false), "less than a minute");
+    assert.equal(timeformat.distanceToNow(seconds, true), "less than a minute ago");
+
+    assert.equal(timeformat.distanceToNow(minutes), "4 minutes");
+    assert.equal(timeformat.distanceToNow(minutes, true), "4 minutes ago");
+
+    assert.equal(timeformat.distanceToNow(halfhour), "32 minutes");
+    assert.equal(timeformat.distanceToNow(halfhour, true), "32 minutes ago");
+
+    assert.equal(timeformat.distanceToNow(hours), "about 4 hours");
+    assert.equal(timeformat.distanceToNow(hours, true), "about 4 hours ago");
+
+    assert.equal(timeformat.distanceToNow(days4short), "4 days");
+    assert.equal(timeformat.distanceToNow(days4long), "4 days");
+    assert.equal(timeformat.distanceToNow(days4short, true), "4 days ago");
+
+    assert.equal(timeformat.distanceToNow(week3short), "20 days");
+    assert.equal(timeformat.distanceToNow(week3long), "21 days");
+    assert.equal(timeformat.distanceToNow(week3long, true), "21 days ago");
+
+    assert.equal(timeformat.distanceToNow(month2short), "about 2 months");
+    assert.equal(timeformat.distanceToNow(month2long), "2 months");
+});
+
+QUnit.test("relative formatter, German", assert => {
+    const now = Date.now();
+    const seconds = now - 4.5 * SEC;
+
+    // no need to be as thorough as with English, just spot check that it's translated
+    cockpit.language = "de";
+    assert.equal(timeformat.distanceToNow(seconds), "weniger als 1 Minute");
+    assert.equal(timeformat.distanceToNow(seconds, true), "vor weniger als 1 Minute");
+
+    assert.equal(timeformat.distanceToNow(now - 4 * DAY), "4 Tage");
+    assert.equal(timeformat.distanceToNow(now - 21 * DAY), "21 Tage");
+    assert.equal(timeformat.distanceToNow(now - 62 * DAY), "2 Monate");
+});
+
+QUnit.test("firstDayOfWeek", assert => {
```

Comment:
ah right.  It would be nice if our coverage reporting also covered the QUnit runs...